vendor/bat-native-ads/include/bat/ads/public/interfaces/ads_database.mojom

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+module ads_database.mojom;
+
+union DBValue {
```

Comment:
Is this all we can do? It seems silly to duplicate this code in 2 places. Couldn't we extract this to a component that both ledger & ads use?

Reply:
I will raise a separate issue to further discuss, thanks

Reply:
There is no mitigation of state as on iOS we were using an in memory bundle state which was reset each time we downloaded the catalog